app/src/main/java/org/dicekeys/app/openpgp/PublicPacket.kt

```diff
@@ -0,0 +1,55 @@
+package org.dicekeys.app.openpgp;
+
+import com.google.common.io.ByteStreams
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+import org.dicekeys.app.extensions.toHex
+import java.math.BigInteger
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+
+class PublicPacket(private val publicKey: ByteArray, private val timestamp: UInt): Packet() {
+
+    override val pTag: Int
+        get() = 0x98
+
+    override val body: ByteArray by lazy {
+        val body = ByteStreams.newDataOutput()
+
+        val taggedPublicKey = byteArrayOf(0x40) + publicKey
+
+        body.write(Version)
+        body.writeInt(timestamp.toInt())
+        body.write(Ed25519Algorithm)
+        body.write(Ed25519CurveOid.size)
+        body.write(Ed25519CurveOid)
+
+        body.write(Mpi.fromByteArray(taggedPublicKey).toByteArray())
+
+        body.toByteArray()
+    }
+
+    fun fingerprint(): ByteArray {
+        val out = ByteStreams.newDataOutput()
+        val digest: MessageDigest = MessageDigest.getInstance("SHA-1")
+
+        out.writeByte(0x99)
```

Comment:
What is 0x99?
Add pointer to spec for it?
It might be more clear if it were stored in a constnat that had a meaningful name (like you've done with pTag in SecurePacket.kt)

Reply:
I am adding documenation, from RFC

> A V4 fingerprint is the 160-bit SHA-1 hash of the octet 0x99,
   followed by the two-octet packet length, followed by the entire
   Public-Key packet starting with the version field.  The Key ID is the
   low-order 64 bits of the fingerprint.